### PR TITLE
End point to list all files from data/settings folder

### LINKED ISSUE #111: Add servlet which lists all files in data/settings/
access must be restricted to admin rights

### PR DESCRIPTION
**Pull request by DravitLochan**
Fixes #111 

Changes: added servlet to list all files from data/settings folder. Requires admin login. 
Test locally at http://127.0.0.1:4000/aaa/getAllFiles
Screenshots for the change: 
image

### CODE REVIEW

**Comment by DravitLochan** [comment]
@dynamitechetan @jyoth1raditya @saurabhjn76 Please review!

**Comment by saurabhjn76** [comment]
@DravitLochan Please also give reference to #111 .

**Comment by DravitLochan** [comment]
@mariobehling Please review and merge!

File: src/ai/susi/server/api/aaa/GetAAAFiles.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return null;
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query post, HttpServletResponse response, Authorization rights, JsonObjectWithDefault permissions) throws APIException {
+
+        String path = DAO.data_dir.getPath()+"/settings/";
+        File settings = new File(path);
+        String[] files = settings.list();
+        System.out.println(files.length);
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: remove this line

File: src/ai/susi/server/api/aaa/GetAAAFiles.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package ai.susi.server.api.aaa;
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: Please add a license header, copy it from the other files and set your own name as copyright holder

File: src/ai/susi/server/api/aaa/GetAAAFiles.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+
+/**
+ * Created by dravit on 14/6/17.
+ * Servlet to get list of all the files present in data/settings directory
+ * test locally at http://127.0.0.1:4000/aaa/getAllFiles
+ */
+public class GetAAAFiles extends AbstractAPIHandler implements APIHandler{
+    @Override
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: This lists settings. The files inside the settings folder are not al about AAA. Please name it ListSettingsService. All Classes which extends AbstractAPIHandler shall be named a "service"

File: src/ai/susi/server/api/aaa/GetAAAFiles.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    public String getAPIPath() {
+        return "/aaa/getAllFiles";
```
Comment (Orbiter) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: name this "/aaa/listSettings"

Comment (Orbiter) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: "/aaa/listSettings.json"

File: src/ai/susi/server/api/aaa/GetAAAFiles.java (hunk 2 above)
Comment (DravitLochan) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: ok

File: src/ai/susi/server/api/aaa/GetAAAFiles.java (hunk 3 above)
Comment (DravitLochan) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: ok

File: src/ai/susi/server/api/aaa/GetAAAFiles.java (hunk 4 above)
Comment (DravitLochan) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: ok

File: src/ai/susi/server/api/aaa/GetAAAFiles.java (hunk 1 above)
Comment (DravitLochan) [inline src/ai/susi/server/api/aaa/GetAAAFiles.java]: ok

**Comment by DravitLochan** [comment]
@Orbiter @mariobehling made the suggested changes. Please review and merge

**Comment by DravitLochan** [comment]
@mariobehling @Orbiter made the changes. Please review and merge so that I can proceed with an application for admins to download these files!
Thanks

**Comment by DravitLochan** [comment]
@Orbiter Please merge it as well. Thanks

**Pull request merged**